### PR TITLE
Update 0x05h-Testing-Platform-Interaction.md

### PR DESCRIPTION
**Pull request by bagipro**
Thank you for submitting a Pull Request to the Mobile Security Testing Guide. Please make sure that:

- [x] Your contribution is written in the 2nd person (e.g. you)
- [x] Your contribution is written in an active present form for as much as possible.
- [x] You have made sure that the reference section is up to date (e.g. please add sources you have used, make sure that the references to MITRE/MASVS/etc. are up to date)
- [x] Your contribution has proper formatted markdown and/or code
- [x] Any references to website have been formatted as [TEXT](URL “NAME”)
- [x] You verified/tested the effectiveness of your contribution (e.g.: is the code really an effective remediation? Please verify it works!)

If your PR is related to an issue. Please end your PR test with the following line:
This PR closes #< insert number here >.

### CODE REVIEW

**Comment by anantshri** [comment]
my only concern: the person in last 48 hours or so have pushing this article in so many places it feels like promotion.

**Comment by bagipro** [comment]
@anantshri 
Just check the articles’ contents! It contains so many techniques that are missing on OWASP

File: Document/0x04h-Testing-Code-Quality.md [hunk 1]
```diff
@@ -222,7 +222,8 @@ Memory corruption bugs are a popular mainstay with hackers. This class of bug re
...
 
-Android apps are, for the most part, implemented in Java which is inherently safe from memory corruption issues by design. However, native apps utilizing JNI libraries are susceptible to this kind of bug.
+Android apps are, for the most part, implemented in Java which is inherently safe from memory corruption issues by design. However, native apps utilizing JNI libraries are susceptible to this kind of bug. In rare cases, Android apps that use XML/JSON parsers to unwrap Java objects are also subject to memory corruption bugs. [An example](https://blog.oversecured.com/Exploiting-memory-corruption-vulnerabilities-on-Android/#example-of-the-vulnerability-in-paypal%E2%80%99s-apps) of such vulnerability was found by Oversecured in the PayPal app.
```
Comment (sushi2k) [inline Document/0x04h-Testing-Code-Quality.md:225]: We would like to stay vendor neutral, therefore can you please remove the company name from the paragraph above?

```suggestion
Android apps are, for the most part, implemented in Java which is inherently safe from memory corruption issues by design. However, native apps utilizing JNI libraries are susceptible to this kind of bug. In rare cases, Android apps that use XML/JSON parsers to unwrap Java objects are also subject to memory corruption bugs. [An example](https://blog.oversecured.com/Exploiting-memory-corruption-vulnerabilities-on-Android/#example-of-the-vulnerability-in-paypal%E2%80%99s-apps) of such vulnerability was found in the PayPal app.
```

File: Document/0x05h-Testing-Platform-Interaction.md [hunk 2]
```diff
@@ -763,6 +763,7 @@ Once you identify a list of IPC mechanisms, review the source code to see whethe
 
 In the following, we use two example apps and give examples of identifying vulnerable IPC components:
 
+- ["OVAA"](https://github.com/oversecured/ovaa "Oversecured Vulnerable Android App")
```
Comment (sushi2k) [inline Document/0x05h-Testing-Platform-Interaction.md:766]: The Oversecured app should be added to the vulnerable app section instead: https://github.com/OWASP/owasp-mstg/blob/master/Document/0x08-Testing-Tools.md#vulnerable-applications

**Review by sushi2k — CHANGES_REQUESTED** [review]
Hi @bagipro, thank you for adding new content! 
Links that are being added to the MSTG should have a value-add to existing content in the MSTG, therefore if attacks, techniques etc are missing it should be added on a high level to the MSTG test case and for further details a link can be added. 

If you think test cases are missing in the MSTG that you are describing in your blog articles, please let us know. Then we could even create new requirements in the MASVS, as we are refactoring it at the moment.

**Comment by bagipro** [comment]
@sushi2k 
Thanks for your feedback. I've made changes.

Regarding the techniques:
1. OWASP docs don't say anything about intent redirection bugs (https://blog.oversecured.com/Android-Access-to-app-protected-components/) and gaining access to content providers with `android:grantUriPermissions="true"` flag set (https://blog.oversecured.com/Gaining-access-to-arbitrary-Content-Providers/). Those vectors are scary because, in combination with dynamic code loading, it leads to persistent LCEs very often (check out bugs in [Google](https://blog.oversecured.com/Why-dynamic-code-loading-could-be-dangerous-for-your-apps-a-Google-example/) and [TikTok](https://blog.oversecured.com/Oversecured-detects-dangerous-vulnerabilities-in-the-TikTok-Android-app/))
2. OWASP section related to implicit intents doesn't say about `startActivityForResult(...)` that can return values via `setResult(...)`. If content pickers are used (such as `android.intent.action.GET_CONTENT` or `android.intent.action.PICK`), it leads to theft and/or overwrite of arbitrary files (in the first case, if the URI's content is saved to a public directory; in the second case if the attacker can control output path via, e.g. `Uri.getLastPathSegments()` or `_display_name` when a vulnerable app queries it from the attacker-controlled content provider)
3. Activities that handle `android.intent.action.SEND` accept a Uri to share. They cache the contents somewhere and then process it as a file. The attacks are the same as in the previous item.
3. Android apps use dynamic code loading too often, and that's an issue. I would be happy if you encourage developers to don't do that.
4. OWASP also doesn't say anything about exposed binders via exported services.
5. From https://blog.oversecured.com/Android-security-checklist-webview/, OWASP doesn't report about UXSS (which is a popular issue too), bypassable URL checks (50/50 that the app validates the URL, but only 10% they are doing it securely) and `WebResourceResponse`
6. https://blog.oversecured.com/Android-arbitrary-code-execution-via-third-party-package-contexts/ this is not covered. But some mostly Chinese apps allow developing plugins for it (such as themes, built-in wallpapers, some action handles). They scan installed apps and automatically invoke some calls to their sources, leading to LCE.

**Comment by bagipro** [comment]
@sushi2k 
Also, look for https://www.blackhat.com/eu-21/briefings/schedule/index.html#re-route-your-intent-for-privilege-escalation-an-universal-way-to-exploit-android-pendingintents-in-high-profile-and-system-apps-24340. They will report a new vector on how to get access to providers via PendingIntents (a lot of apps are vulnerable to that right now!)

**Comment by cpholguera** [comment]
Unable to merge, replacing it with #2284
Comments and other suggestions from the discussion above will be included as Issues.

Thank you!

**Pull request closed without merging**